dataland-batch-manager/src/main/kotlin/org/dataland/datalandbatchmanager/service/GleifGoldenCopyIngestor.kt

```diff
@@ -65,22 +67,33 @@ class GleifGoldenCopyIngestor(
 
             logger.info("Retrieving all company data available via GLEIF.")
             val tempFile = File.createTempFile("gleif_golden_copy", ".csv")
-            processFile(tempFile, gleifApiAccessor::getFullGoldenCopy)
+            processDeltaFile(tempFile, gleifApiAccessor::getFullGoldenCopy)
+            if (savedMappingFile.exists()) {
+                if (!savedMappingFile.delete()) {
```

Comment:
We should not try to delete it because it should not be there in the first place. Also, rather throw an error if it already exists.

Reply:
Why not? What if the file exists due to last month's update, then somebody decides to activate the fullGoldenCopy.

we have this code later which will run much faster if we delete the savedMappingFile for a goldenCopy

val deltaMap: Map<String, String> = if (!savedMappingFile.exists() || savedMappingFile.length() == 0L) {
            isinDeltaBuilder.createDeltaOfMappingFile(newMappingFileCsv, null)
        } else {
            isinDeltaBuilder.createDeltaOfMappingFile(newMappingFileCsv, savedMappingFile)
        }

Reply:
Isn't the file deleted anyways if we redeploy? And we can only trigger a new full copy at redeployment anyways. But I'm not sure about this.

Reply:
Aah, you are preserving them via volumes. I need to further check that.